sprint10

andvatar opened the pull request.

andvatar committed:
sprint 10 initial commit

andvatar committed:
sprint 10 initial commit

andvatar committed:
Merge remote-tracking branch 'origin/sprint10' into sprint10

andvatar committed:
Merge remote-tracking branch 'origin/sprint10' into sprint10

andvatar committed:
Merge remote-tracking branch 'origin/sprint10' into sprint10

andvatar committed:
Merge remote-tracking branch 'origin/sprint10' into sprint10

andvatar committed:
Merge remote-tracking branch 'origin/sprint10' into sprint10

andvatar committed:
Merge remote-tracking branch 'origin/sprint10' into sprint10

andvatar committed:
Merge remote-tracking branch 'origin/sprint10' into sprint10

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint10 fixes

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

Perederey commented on bank-app/values.yaml:
```diff
@@ -0,0 +1,260 @@
+  # Database connection details passed to customer-service configmap
+  config:
+    # --- CORRECTED DB URL --- Using '<ReleaseName>-<Alias>' service name convention
+    database_url: "jdbc:postgresql://{{ .Release.Name }}-bank-db.{{ .Release.Namespace }}.svc.cluster.local:5432/bank-db"
```
Вот тут я бы добавил больше параметризации для database_url - сейчас жестко прописан шаблон, а хорошо бы было вынести имя БД, порт и схему в отдельные переменные, чтобы можно было легко переопределить для разных окружений

Perederey commented on bank-app/templates/configmap.yaml:
```diff
@@ -0,0 +1,2660 @@
+        "policies": [
+        ]
+      }
+    }
```
Намудрил тут немного с огромным JSON-конфигом прямо в ConfigMap - это нарушает принцип разделения конфигурации и делает чарт сложным для поддержки

Perederey commented on accounts/src/main/resources/application.yaml:
```diff
@@ -3,22 +3,22 @@ spring:
-      config:
-        enabled: false
-        defaultContext: application
+    url: ${SPRING_DATASOURCE_URL:jdbc:postgresql://localhost:5433/yandex_bank}?currentSchema=accounts
```
закомментированный код лучше было бы удалить совсем, он только засоряет конфиг

Perederey replied:
вот так можно

```yaml
spring:
  datasource:
    url: ${SPRING_DATASOURCE_URL:jdbc:postgresql://${DB_HOST:localhost}:${DB_PORT:5433}/${DB_NAME:yandex_bank}}?currentSchema=${DB_SCHEMA:accounts}
    username: ${SPRING_DATASOURCE_USERNAME:postgres}
    password: ${SPRING_DATASOURCE_PASSWORD:postgres}
```

Perederey commented on bank-app/charts/accounts/templates/deployment.yaml:
```diff
@@ -0,0 +1,93 @@
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
```
Вот тут я бы сделал так - вынес бы общий template для всех микросервисов, поскольку deployment-ы очень похожи:

```yaml
{{- define "microservice.deployment" -}}
apiVersion: apps/v1
kind: Deployment
metadata:
  name: {{ include (printf "%s.fullname" .Chart.Name) . }}
  labels:
    {{- include (printf "%s.labels" .Chart.Name) . | nindent 4 }}
spec:
  template:
    spec:
      containers:
        - name: {{ .Chart.Name }}
          env:
            {{- include "microservice.commonEnv" . | nindent 12 }}
            {{- include (printf "%s.specificEnv" .Chart.Name) . | nindent 12 }}
{{- end -}}
```

Perederey commented on bank-app/templates/configmap.yaml:
```diff
@@ -0,0 +1,2660 @@
+  name: keycloak-realm-config
+  namespace: {{ .Release.Namespace }}
+data:
+  yandex-bank.json: |
```
вот типа такого

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: keycloak-realm-config
data:
  {{- (.Files.Glob "keycloak/*.json").AsConfig | nindent 2 }}
```

Perederey commented on exchange/src/main/java/ru/yandex/practicum/tarasov/exchange/controller/ExchangeController.java:
```diff
@@ -29,7 +29,7 @@ public long exchangeCurrencies(@PathVariable("amount") long amount,
     }
 
     @GetMapping("/rates")
-    @CrossOrigin(origins = "http://localhost:8090")
```
можно так

```java
@Configuration
@EnableWebMvc
public class CorsConfig implements WebMvcConfigurer {
    @Override
    public void addCorsMappings(CorsRegistry registry) {
        registry.addMapping("/rates")
                .allowedOrigins("http://front-ui.test.local", "http://front-ui.prod.local")
                .allowedMethods("GET");
    }
}
```

Perederey commented on frontUI/src/main/resources/templates/main.html:
```diff
@@ -6,7 +6,7 @@
         setInterval(() => {
             var td = document.getElementById('exchange_rates');
-            fetch('http://localhost:8084/rates')
+            fetch('http://exchange.test.local/rates')
```
тут бы лучше убрать хардкод URL

Perederey commented on jenkins/Jenkinsfile:
```diff
@@ -0,0 +1,396 @@
+                        --namespace test --create-namespace \\
+                        --set global.domain=test.local \\
+                        --set accounts.image.tag=${IMAGE_TAG} \\
+                        --set accounts.ingress.hosts[0].host=accounts.test.local \\
```
можно использовать values.yaml файла вместо кучи --set флагов

Perederey commented on accounts/Dockerfile:
```diff
@@ -6,7 +6,7 @@ RUN --mount=type=cache,target=/root/.m2 mvn clean package -DskipTests
 WORKDIR /app
 
-COPY --from=builder /app/accounts/target/*.jar app.jar
+COPY --from=builder app/target/accounts-*.jar app.jar
```
имя можно было бы сделать более специфичным для избежания конфликтов при сборке

Perederey commented on jenkins/Jenkinsfile:
```diff
@@ -0,0 +1,396 @@
+            steps {
+                dir('bank-app') {
+                    sh "helm dependency update"
+                    sh "helm dependency build"
```
Вижу, что ты дублируешь почти одинаковые команды helm upgrade для test и prod, но было бы лучше вынести общие параметры в переменные или использовать функции

Perederey reviewed (commented):
Привет! Отлично справился с переводом приложения в k8s!

Из того, на что стоит обратить внимание:
- Много дублирования в конфигурациях Helm чартов - можно вынести общие части в templates или global секцию
- Hardcoded URLы и другие значения лучше параметризовать для большей гибкости между окружениями
- Огромный ConfigMap с JSON конфигом Keycloak стоит разбить на более управляемые части

А так молодец! Архитектура получилась солидная, и Jenkins pipeline выглядит профессионально!

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 final

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 fixing deployments

andvatar committed:
sprint 10 feign clients fix

andvatar committed:
sprint 10 feign clients fix

andvatar committed:
исправления:
Добавил дополнительные параметры к db_url
Вынес json-конфиг в отдельный файл
Сделал шаблоны для генерации deployments и ingresses
Добавил параметр для запроса курсов валют
Убрал лишние --set из helm upgrade --install. К ingress.hosts.host добавляю test.local или prod.local при генерации ingress
Переименовал сервисы для избежания конфликтов

andvatar committed:
исправления:
Добавил дополнительные параметры к db_url
Вынес json-конфиг в отдельный файл
Сделал шаблоны для генерации deployments и ingresses
Добавил параметр для запроса курсов валют
Убрал лишние --set из helm upgrade --install. К ingress.hosts.host добавляю test.local или prod.local при генерации ingress
Переименовал сервисы для избежания конфликтов
